x/yarpctest/request_stream.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package yarpctest
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/transport/grpc"
```

Comment:
Is this right? `yarpctest` relying on individual transports.

Reply:
Yup, we're already doing this for the existing grpc/tchannel/http Requests and Services.  There's not much of a way around it unfortunately, we could pull all those functions into different packages, but that might be a bit overkill.

In general if we have a circular dependency on where we want to use these packages the solution will be to use the **\<package\>_test** package instead of the main one (this is what we did for the stream_test.go in the grpc folder.

Reply:
I would not mind yarpc/yarpctest/yarpchttptest to keep this concern separable. We should at least try to pretend yarpc is an API-only package with support from yarpc-plugin packages.

Reply:
I am really flexible here, if you want to refactor go for it, otherwise I leave enforcement up to @kriskowal heh